src/OSPSuite.SimModel/src/SimpleProductFormula.cpp

```diff
@@ -110,6 +110,13 @@ void SimpleProductFormula::SetQuantityReference (const QuantityReference & quant
 {
 	_quantityRefs.push_back(quantityReference);
 	UpdateFromQuantityReference(quantityReference);
+
+	//If the quantity is a species, add this formula to the list of entities that cache its scale factor (used for updating scale factors)
+	if (quantityReference.IsSpecies())
+	{
+		Species * species = quantityReference.GetSpecies();
+		species->AddEntityWithCachedScaleFactor(this);
```

Comment:
Save ref to the current formula in the species's list of entities which cache it's scale factor